baggage-processor/src/main/java/io/opentelemetry/contrib/baggage/processor/BaggageSpanProcessorCustomizer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.baggage.processor;
+
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import java.util.List;
+
+public class BaggageSpanProcessorCustomizer implements AutoConfigurationCustomizerProvider {
+  @Override
+  public void customize(AutoConfigurationCustomizer autoConfigurationCustomizer) {
+    autoConfigurationCustomizer.addTracerProviderCustomizer(
+        (sdkTracerProviderBuilder, config) -> {
+          addSpanProcessor(sdkTracerProviderBuilder, config);
+          return sdkTracerProviderBuilder;
+        });
+  }
+
+  private static void addSpanProcessor(
+      SdkTracerProviderBuilder sdkTracerProviderBuilder, ConfigProperties config) {
+    List<String> keys =
+        config.getList("otel.java.experimental.span-attributes.copy-from-baggage.include");
+
+    if (keys.isEmpty()) {
+      return;
+    }
+
+    sdkTracerProviderBuilder.addSpanProcessor(createProcessor(keys));
+  }
+
+  static BaggageSpanProcessor createProcessor(List<String> keys) {
+    if (keys.size() == 1 && keys.get(0).equals("*")) {
+      return new BaggageSpanProcessor(BaggageSpanProcessor.allowAllBaggageKeys);
```

Comment:
unrelated, I'd suggest turning `allowAllBaggageKeys` into a static method returning `BaggageSpanProcessor`, which matches the pattern we've used elsewhere, e.g. https://github.com/open-telemetry/opentelemetry-java/blob/4680146024b83892c14e5df4e08158f1f602c75c/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java#L29

and then you can do the simpler here:

```suggestion
      return BaggageSpanProcessor.allowAllBaggageKeys();
```

Reply:
I like it - added and deprecated the old field